File: surface/widgets/compose/components/widgets/data-display/NotificationItem.kt
// ============================================================
// Clef Surface Compose Widget — NotificationItem
//
// Single notification entry displaying a title, optional
// description, timestamp, and read/unread state. Compose
// adaptation: Row with leading unread dot indicator, icon,
// title/description Column, trailing timestamp, and clickable
// surface with swipe-to-dismiss support.
// See widget spec: repertoire/widgets/data-display/notification-item.widget
// ============================================================

package clef.surface.compose.components.widgets.datadisplay

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp

// --------------- Component ---------------

/**
 * Single notification entry with read/unread state, icon, and timestamp.
 *
 * @param title Primary notification text.
 * @param description Optional secondary detail text.
 * @param timestamp Timestamp string (relative or absolute).
 * @param read Whether the notification has been read.
 * @param icon Optional leading icon.
 * @param onPress Callback when the notification is tapped.
 * @param onDismiss Callback when the notification is dismissed.
 * @param modifier Modifier for the root layout.
 */
@Composable
fun NotificationItem(
    title: String,
    description: String? = null,
    timestamp: String? = null,
    read: Boolean = false,
    icon: ImageVector? = null,
    onPress: (() -> Unit)? = null,
    onDismiss: (() -> Unit)? = null,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .then(
                if (onPress != null) Modifier.clickable(onClick = onPress) else Modifier,
            )
            .padding(horizontal = 16.dp, vertical = 12.dp),
        verticalAlignment = Alignment.Top,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        // Unread indicator dot
        Box(
            modifier = Modifier
                .padding(top = 6.dp)
                .size(8.dp)
                .clip(CircleShape)
                .background(
                    if (!read) {
                        MaterialTheme.colorScheme.primary
                    } else {
                        MaterialTheme.colorScheme.surface
                    },
                ),
        )

        // Icon
        if (icon != null) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = MaterialTheme.colorScheme.onSurfaceVariant,
                modifier = Modifier
                    .size(24.dp)
                    .padding(top = 2.dp),
            )
        }

        // Title and description
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = title,
                style = MaterialTheme.typography.bodyMedium,
                fontWeight = if (!read) FontWeight.Bold else FontWeight.Normal,
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
            )
            if (description != null) {
                Spacer(modifier = Modifier.height(2.dp))
                Text(
                    text = description,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis,
                )
            }
        }

        // Timestamp
        if (timestamp != null) {
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = timestamp,
                style = MaterialTheme.typography.labelSmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}

File: surface/widgets/swiftui/components/widgets/data-display/NotificationItem.swift
// ============================================================
// Clef Surface SwiftUI Widget — NotificationItem
//
// Single notification entry with icon, title, description,
// timestamp, and read/unread state.
// ============================================================

import SwiftUI

// --------------- Component ---------------

/// NotificationItem view for a single notification entry.
///
/// - Parameters:
///   - title: Primary notification text.
///   - description: Optional secondary text.
///   - timestamp: Timestamp string for the notification.
///   - read: Whether the notification has been read.
///   - icon: SF Symbol name for the notification icon.
///   - onTap: Callback when the notification is tapped.
struct NotificationItemView: View {
    var title: String
    var description: String? = nil
    var timestamp: String? = nil
    var read: Bool = false
    var icon: String = "bell.fill"
    var onTap: (() -> Void)? = nil

    var body: some View {
        SwiftUI.Button(action: { onTap?() }) {
            HStack(alignment: .top, spacing: 12) {
                Image(systemName: icon)
                    .foregroundColor(read ? .secondary : .accentColor)
                    .frame(width: 24, height: 24)

                VStack(alignment: .leading, spacing: 4) {
                    Text(title)
                        .font(.subheadline)
                        .fontWeight(read ? .regular : .semibold)
                        .foregroundColor(.primary)

                    if let description = description {
                        Text(description)
                            .font(.caption)
                            .foregroundColor(.secondary)
                    }

                    if let timestamp = timestamp {
                        Text(timestamp)
                            .font(.caption2)
                            .foregroundColor(.secondary)
                    }
                }

                Spacer()

                if !read {
                    Circle()
                        .fill(Color.accentColor)
                        .frame(width: 8, height: 8)
                }
            }
            .padding(.vertical, 8)
        }
        .buttonStyle(.plain)
        .accessibilityLabel(title)
    }
}
